web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/SingleUseKeyServiceTest.java

```diff
@@ -0,0 +1,159 @@
+    private SingleUseKeyService singleUseKeyService;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        csvPath = tmpFolder.resolve("singleUseKey");
+        singleUseKeyService = new SingleUseKeyService();
+        singleUseKeyService.setCsvPath(csvPath);
+    }
+
+    @Test
+    public void testGenerateNoCsvExists() throws IOException {
+        var key = singleUseKeyService.generate(UUID_TEST);
+        var newRecords = parseCsv(CSV_HEADERS, csvPath);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertCsvContainsCorrectEntry(newRecords, UUID_TEST, key, currentMilliseconds);
+    }
+
+    @Test
+    public void testGenerateMultipleCallsForDifferentIds() throws IOException {
+        var key1 = singleUseKeyService.generate(UUID_1);
+        var key2 = singleUseKeyService.generate(UUID_2);
+        var key3 = singleUseKeyService.generate(UUID_3);
+
+        var newRecords = parseCsv(CSV_HEADERS, csvPath);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertCsvContainsCorrectEntry(newRecords, UUID_1, key1, currentMilliseconds);
+        assertCsvContainsCorrectEntry(newRecords, UUID_2, key2, currentMilliseconds);
+        assertCsvContainsCorrectEntry(newRecords, UUID_3, key3, currentMilliseconds);
+    }
+
+    @Test
+    public void testGenerateMultipleCallsForSameId() throws IOException {
+        var key1 = singleUseKeyService.generate(UUID_1);
+        var key2 = singleUseKeyService.generate(UUID_1);
+        var key3 = singleUseKeyService.generate(UUID_1);
+
+        var newRecords = parseCsv(CSV_HEADERS, csvPath);
+        var currentMilliseconds = System.currentTimeMillis();
+        assertCsvContainsCorrectEntry(newRecords, UUID_1, key1, currentMilliseconds);
+        assertCsvContainsCorrectEntry(newRecords, UUID_1, key2, currentMilliseconds);
+        assertCsvContainsCorrectEntry(newRecords, UUID_1, key3, currentMilliseconds);
+
+    }
+
+    @Test
+    public void testKeyIsValid() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var futureExpiration = System.currentTimeMillis() + DAY_MILLISECONDS;
+        generateDefaultCsv(key, futureExpiration);
+        assertTrue(singleUseKeyService.keyIsValid(key));
+    }
+
+    @Test
+    public void testKeyIsNotValid() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var timestamp = System.currentTimeMillis();
+        generateDefaultCsv(null, timestamp);
+        assertFalse(singleUseKeyService.keyIsValid(key));
+    }
+
+    @Test
+    public void testKeyIsNotValidCurrentTimeIsMoreThan24hLater() throws IOException {
+        var key = SingleUseKeyService.getKey();
+        var pastTimestamp = System.currentTimeMillis() - (2 * DAY_MILLISECONDS);
+        generateDefaultCsv(key, pastTimestamp);
+        assertFalse(singleUseKeyService.keyIsValid(key));
+    }
+
+    @Test
+    public void testInvalidate() throws IOException {
```

Comment:
We should add assertions about the number of entries into all the invalidate tests too, and also add a test that invalidates 2 keys in a row to make sure it doesn't run into any weirdness with the headers.